hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1269,6 +1269,9 @@ private void startServiceThreads() throws IOException {
       HConstants.MASTER_LOG_REPLAY_OPS_THREADS, HConstants.MASTER_LOG_REPLAY_OPS_THREADS_DEFAULT));
     this.service.startExecutorService(ExecutorType.MASTER_SNAPSHOT_OPERATIONS, conf.getInt(
       SnapshotManager.SNAPSHOT_POOL_THREADS_KEY, SnapshotManager.SNAPSHOT_POOL_THREADS_DEFAULT));
+    this.service.startExecutorService(ExecutorType.MASTER_MERGE_OPERATIONS, conf.getInt(
```

Comment:
Are we starting up two threads here that will be sitting around doing nothing for 99.9999999% of the time? Does the executor let all threads die if not being used?

Reply:
Normally it does, but it looks like we are using corePoolSize = maxThreads [1]. Per [documentation](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html), eviction of unused threads (see keepAliveTime doc) only happens if corePoolSize < maxThreads. It looks like we can force core threads too time out too with [allowCoreThreadTimeout](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html#allowCoreThreadTimeOut(boolean))

Mind if I submit a follow up patch that makes these configurable along with the appropriate defaults that apply to all executors?


[1] https://github.com/apache/hbase/blame/master/hbase-server/src/main/java/org/apache/hadoop/hbase/executor/ExecutorService.java#L203

Reply:
@saintstack created #2922 lets move this discussion to that PR.